postgres/tests/test_discovery.py

```diff
@@ -0,0 +1,171 @@
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+import copy
+import time
+from contextlib import contextmanager
+
+import psycopg2
+import psycopg2.sql
+import pytest
+
+from datadog_checks.base import ConfigurationError
+
+from .common import HOST, PASSWORD_ADMIN, USER_ADMIN, _get_expected_tags
+from .utils import requires_over_13, run_one_check
+
+DISCOVERY_CONFIG = {
+    "enabled": True,
+    "include": ["dogs_([1-9]|[1-9][0-9]|10[0-9])"],
+    "exclude": ["dogs_5$", "dogs_50$"],
+}
+
+RELATION_METRICS = {
+    'postgresql.seq_scans',
+    'postgresql.seq_rows_read',
+    'postgresql.rows_inserted',
+    'postgresql.rows_updated',
+    'postgresql.rows_deleted',
+    'postgresql.rows_hot_updated',
+    'postgresql.live_rows',
+    'postgresql.dead_rows',
+    'postgresql.heap_blocks_read',
+    'postgresql.heap_blocks_hit',
+    'postgresql.vacuumed',
+    'postgresql.autovacuumed',
+    'postgresql.analyzed',
+    'postgresql.autoanalyzed',
+}
+
+
+@pytest.mark.integration
+@pytest.mark.usefixtures('dd_environment')
+def test_autodiscovery_simple(integration_check, pg_instance):
+    """
+    Test simple autodiscovery.
+    """
+    pg_instance["database_autodiscovery"] = DISCOVERY_CONFIG
+    pg_instance['relations'] = ['pg_index']
+    del pg_instance['dbname']
+    check = integration_check(pg_instance)
+    run_one_check(check, pg_instance)
+
+    assert check.autodiscovery is not None
+    databases = check.autodiscovery.get_items()
+    expected_len = 100 - len(DISCOVERY_CONFIG["exclude"])
```

Comment:
nit: please comment with an explanation where [magic constants](https://en.wikipedia.org/wiki/Magic_number_(programming)) come from. I know we don't always do that but that's a pattern I'd like to establish in this codebase.